pkg/endpointslice/controller_test.go

```diff
@@ -0,0 +1,160 @@
+				t.createEndpointSlice(testNS1, endpointSlice)
+
+				esName2 := testName2 + remoteClusterID2
+				endPoint2 := t.newEndpoint(cluster2HostNamePod1, cluster2EndPointIP1)
+				endpointSlice2 := t.newEndpointSliceFromEndpoint(testService2, remoteClusterID2, esName2, testNS2, []v1beta1.Endpoint{endPoint2})
+				t.createEndpointSlice(testNS2, endpointSlice2)
+
+				t.awaitIsHealthy(testService1, testNS1, remoteClusterID1, false)
+				t.awaitIsHealthy(testService2, testNS2, remoteClusterID2, false)
+			})
+		})
+	})
+
+	When("IsHealthy is called for a non-existent cluster", func() {
+		It("should return false", func() {
+			esName1 := testName1 + remoteClusterID1
+			endPoint1 := t.newEndpoint(cluster1HostNamePod1, cluster1EndPointIP1)
+			endpointSlice := t.newEndpointSliceFromEndpoint(testService1, remoteClusterID1, esName1, testNS1, []v1beta1.Endpoint{endPoint1})
+			t.createEndpointSlice(testNS1, endpointSlice)
+
+			t.awaitIsHealthy(testService1, testNS1, "randomcluster", true)
+		})
+	})
+})
+
+type endpointSliceTestDriver struct {
+	controller *endpointslice.Controller
+	kubeClient kubernetes.Interface
+	epMap      *endpointslice.Map
+}
+
+func newEndpointSliceTestDiver() *endpointSliceTestDriver {
+	t := &endpointSliceTestDriver{}
+
+	BeforeEach(func() {
+		t.kubeClient = fakeKubeClient.NewSimpleClientset()
+		t.epMap = endpointslice.NewMap()
+		t.controller = endpointslice.NewController(t.epMap)
+		t.controller.NewClientset = func(c *rest.Config) (kubernetes.Interface, error) {
+			return t.kubeClient, nil
+		}
+
+		Expect(t.controller.Start(&rest.Config{})).To(Succeed())
+	})
+
+	AfterEach(func() {
+		t.controller.Stop()
+	})
+
+	return t
+}
+
+func (t *endpointSliceTestDriver) createEndpointSlice(namespace string, endpointSlice *v1beta1.EndpointSlice) {
+	_, err := t.kubeClient.DiscoveryV1beta1().EndpointSlices(namespace).Create(endpointSlice)
+	Expect(err).To(Succeed())
+}
+
+func (t *endpointSliceTestDriver) newEndpointSliceFromEndpoint(serviceName, remoteCluster, esName,
+	esNs string, endPoints []v1beta1.Endpoint) *v1beta1.EndpointSlice {
+	return &v1beta1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      esName,
+			Namespace: esNs,
+			Labels: map[string]string{
+				v1beta1.LabelManagedBy:           lhconstants.LabelValueManagedBy,
+				lhconstants.LabelSourceName:      serviceName,
+				lhconstants.LabelSourceCluster:   remoteCluster,
+				lhconstants.LabelSourceNamespace: esNs,
+			},
+		},
+		Endpoints: endPoints,
+	}
+}
+
+func (t *endpointSliceTestDriver) newEndpoint(hostname, ipaddress string) v1beta1.Endpoint {
+	return v1beta1.Endpoint{
+		Addresses:  []string{ipaddress},
+		Conditions: v1beta1.EndpointConditions{},
+		Hostname:   &hostname,
+	}
+}
+
+func (t *endpointSliceTestDriver) awaitIsHealthy(name, nameSpace, clusterId string, fail bool) {
+	if fail {
+		Eventually(func() bool {
+			return t.controller.IsHealthy(name, nameSpace, clusterId)
+		}, 5).Should(BeFalse())
```

Comment:
Move to `awaitNotIsHealthy`.

```suggestion
		Consistently(func() bool {
			return t.controller.IsHealthy(name, nameSpace, clusterId)
		}, 500*time.Millisecond).Should(BeFalse())
```